src/extract.rs

```diff
@@ -11,13 +11,45 @@ use futures_core::ready;
 
 use crate::{dev::Payload, Error, HttpRequest};
 
-/// Trait implemented by types that can be extracted from request.
+/// A type that implements [`FromRequest`] is called an **extractor** and can extract data
+/// from the request. Examples of types that implement this trait are [`Json`], [`Form`], [`Path`].
 ///
-/// Types that implement this trait can be used with `Route` handlers.
+/// An extractor can be customized by injecting the corresponding configuration with one of:
+///
+/// - [`App::data()`](`crate::App::data`)
```

Comment:
Given that `::data` has been marked as deprecated in favour of `::app_data` in the main branch it probably shouldn't be recommended here as an option.